### PR TITLE
ARROW-14033: [Ruby][Doc] Add macOS development guide for Red Arrow

### PR DESCRIPTION
**Pull request by okadakk**

### CODE REVIEW

**Comment by kou** [comment]
Could you move this to `ruby/red-arrow/README.md` because this describes only Red Arrow?
If we add this to `ruby/README.md`, we should describe other gems such as Red Parquet too.

**Comment by okadakk** [comment]
I made a mistake...
I moved this to ruby/red-arrow/README.md!

File: ruby/red-arrow/README.md [hunk 1]
```diff
@@ -50,3 +50,20 @@ table = Arrow::Table.load("/dev/shm/data.arrow")
 # Process data in table
 table.save("/dev/shm/data-processed.arrow")
 ```
+
+## development
```
Comment (kou) [inline ruby/red-arrow/README.md:54]:
```suggestion
## Development

Note that you need to install Apache Arrow C++/GLib at master before preparing Red Arrow. See also:

  * For Apache Arrow C++: https://arrow.apache.org/docs/developers/cpp/building.html
  * For Apache Arrow GLib: https://github.com/apache/arrow/blob/master/c_glib/README.md

```

File: ruby/red-arrow/README.md [hunk 2]
```diff
@@ -50,3 +50,20 @@ table = Arrow::Table.load("/dev/shm/data.arrow")
...
+```
```
Comment (kou) [inline ruby/red-arrow/README.md:55]:
```suggestion
```console
```

File: ruby/red-arrow/README.md [hunk 3]
```diff
@@ -50,3 +50,20 @@ table = Arrow::Table.load("/dev/shm/data.arrow")
...
+$ cd ruby/red-arrow
+$ bundle install
+$ bundle exec rake test
+```
+
+### Mac OS
```
Comment (kou) [inline ruby/red-arrow/README.md:61]:
```suggestion
### For macOS with Homebrew

```

File: ruby/red-arrow/README.md [hunk 4]
```diff
@@ -50,3 +50,20 @@ table = Arrow::Table.load("/dev/shm/data.arrow")
...
+```
```
Comment (kou) [inline ruby/red-arrow/README.md:62]:
```suggestion
```console
```

File: ruby/red-arrow/README.md [hunk 5]
```diff
@@ -50,3 +50,20 @@ table = Arrow::Table.load("/dev/shm/data.arrow")
...
+$ cd ruby/red-arrow
+$ bundle install
+$ brew install apache-arrow --head
+$ brew install apache-arrow-glib --head
+$ export PKG_CONFIG_PATH="/opt/homebrew/opt/openssl@1.1/lib/pkgconfig"
```
Comment (kou) [inline ruby/red-arrow/README.md:67]: Can we remove this with the following patch?

```diff
diff --git a/ruby/red-arrow/ext/arrow/extconf.rb b/ruby/red-arrow/ext/arrow/extconf.rb
index 5ba9f4c2e..9e92bd316 100644
--- a/ruby/red-arrow/ext/arrow/extconf.rb
+++ b/ruby/red-arrow/ext/arrow/extconf.rb
@@ -25,6 +25,19 @@ if arrow_pkg_config_path
   ENV["PKG_CONFIG_PATH"] = pkg_config_paths.join(File::PATH_SEPARATOR)
 end
 
+checking_for(checking_message("Homebrew")) do
+  platform = NativePackageInstaller::Platform.detect
+  if platform.is_a?(NativePackageInstaller::Platform::Homebrew)
+    openssl_prefix = `brew --prefix openssl@1.1`.chomp
+    unless openssl_prefix.empty?
+      PKGConfig.add_path("#{openssl_prefix}/lib/pkgconfig")
+    end
+    true
+  else
+    false
+  end
+end
+
 unless required_pkg_config_package([
                                      "arrow",
                                      Arrow::Version::MAJOR,
```

**Review by kou — APPROVED** [review]
+1

File: ruby/red-arrow/README.md (hunk 5 above)
Comment (kou) [reply ruby/red-arrow/README.md:74]: BTW, could you show your `$(brew --prefix)/lib/pkgconfig/arrow*.pc`?

**Comment by okadakk** [comment]
Thanks!!

File: ruby/red-arrow/README.md (hunk 5 above)
Comment (okadakk) [reply ruby/red-arrow/README.md:74]: @kou
Thank you for reviewing! 

I added the following patch and removed the comment.
https://github.com/apache/arrow/pull/11188
This change worked correctly in my environment!

> BTW, could you show your $(brew --prefix)/lib/pkgconfig/arrow*.pc?

```
> ls $(brew --prefix)/lib/pkgconfig/arrow*.pc

/opt/homebrew/lib/pkgconfig/arrow-compute.pc            /opt/homebrew/lib/pkgconfig/arrow-glib.pc
/opt/homebrew/lib/pkgconfig/arrow-csv.pc                /opt/homebrew/lib/pkgconfig/arrow-json.pc
/opt/homebrew/lib/pkgconfig/arrow-dataset-glib.pc       /opt/homebrew/lib/pkgconfig/arrow-orc-glib.pc
/opt/homebrew/lib/pkgconfig/arrow-dataset.pc            /opt/homebrew/lib/pkgconfig/arrow-orc.pc
/opt/homebrew/lib/pkgconfig/arrow-filesystem.pc         /opt/homebrew/lib/pkgconfig/arrow-python-flight.pc
/opt/homebrew/lib/pkgconfig/arrow-flight-glib.pc        /opt/homebrew/lib/pkgconfig/arrow-python.pc
/opt/homebrew/lib/pkgconfig/arrow-flight.pc             /opt/homebrew/lib/pkgconfig/arrow.pc
```

Comment (kou) [reply ruby/red-arrow/README.md:74]: Thanks.

Could you also show test contents of them?

**Pull request closed without merging**